x-pack/plugins/data_enhanced/public/search/sessions_mgmt/components/table/table.tsx

```diff
@@ -53,41 +47,40 @@ export function SearchSessionsMgmtTable({
     [isLoading]
   );
 
-  // Convert the config interval to ms
-  useEffect(() => {
-    setRefreshInterval(moment.duration(config.refreshInterval).asMilliseconds());
-  }, [config.refreshInterval]);
-
   // refresh behavior
   const doRefresh = useCallback(async () => {
     setIsLoading(true);
+    const renderResults = (results: UISession[]) => {
+      setTableData(results);
+    };
+    showLatestResultsHandler.current = renderResults;
     try {
-      const renderResults = (results: UISession[]) => {
-        setTableData(results);
-      };
-      showLatestResultsHandler.current = renderResults;
       const results = await api.fetchTableData();
-      if (results) {
-        // Ignore results from any but the latest api call
-        if (showLatestResultsHandler.current === renderResults) renderResults(results);
-      } else {
-        renderResults([]);
+
+      if (showLatestResultsHandler.current === renderResults) {
+        renderResults(results || []);
+        setIsLoading(false);
       }
     } catch (e) {
       setTableData([]);
```

Comment:
We should also check `if (showLatestResultsHandler.current === renderResults)` here before updating the state. it could be that we are no longer interested in this result.